testnet/counter.js

```diff
@@ -13,30 +13,38 @@ const { privateKey } = require('../privateKey');
         const FEE = 1500;
         
         // lock fund to the script
-        const lockingTx =  await createLockingTx(privateKey.toAddress(), amount, counter.lockingScript)
-        lockingTx.sign(privateKey)
-        let lockingTxid = await sendTx(lockingTx)
-        console.log('funding txid:      ', lockingTxid)
+        const lockingTx =  await deployContract(counter, amount)
+        console.log('funding txid:      ', lockingTx.id);
+
+        let prevTx = lockingTx;
 
         // unlock
         for (i = 0; i < 3; i++) {
-            let prevLockingScript = counter.lockingScript;
             
-            // update state
-            counter.setDataPart(num2bin(i + 1, DataLen))
-            const newLockingScript = counter.lockingScript;
-            const newAmount = amount - FEE
+            const newState = num2bin(i + 1, DataLen);
 
-            const unlockingTx = await createUnlockingTx(lockingTxid, amount, prevLockingScript, newAmount, newLockingScript)
+            const newLockingScript = bsv.Script.fromASM([counter.codePart.toASM(), newState].join(' '))
+            const newAmount = amount - FEE
 
-            const preimage = getPreimage(unlockingTx, prevLockingScript, amount)
-            const unlockingScript = counter.increment(new SigHashPreimage(toHex(preimage)), newAmount).toScript()
-            unlockingTx.inputs[0].setScript(unlockingScript)
+            const unlockingTx = new bsv.Transaction();
+            
+            unlockingTx.addInput(createInputFromTx(prevTx))
+            .addOutput(new bsv.Transaction.Output({
+                script: newLockingScript,
+                satoshis: newAmount,
+              }))
+            .setInputScript(0, (tx, output) => {
+                const preimage = getPreimage(tx, output.script, output.satoshis)
```

Comment:
This requested change hasn't been made:

getPreimage(tx, Output)

Reply:
need to change getPreimage function